ddev/src/ddev/cli/ci/codeowners.py

```diff
@@ -0,0 +1,70 @@
+import click
+from datadog_checks.dev.tooling.codeowners import CodeOwners
+
+from ddev.cli.application import Application
+
+
+@click.command()
+@click.option('--pr', help='Check the codeowners for the given PR')
+@click.option('--sha', help='Check the codeowners for the given sha')
+@click.option('--files', help='Check the codeowners for the given files separated by commas')
+@click.option('--per-file', is_flag=True, help='Output the codeowners for each file')
+@click.pass_obj
+def codeowners(app: Application, pr: str, sha: str, files: str, per_file: bool):
+    """
+    Check the codeowners for the given PR, commit, or files.
+    """
+    codeowners_file = app.repo.path / '.github' / 'CODEOWNERS'
+    with open(codeowners_file, 'r') as f:
+        codeowners_content = f.readlines()
+    codeowners: CodeOwners = CodeOwners("\n".join(codeowners_content))
+
+    if files:
+        file_list = files.split(',') if ',' in files else [files]
+    elif pr:
+        github = app.github
+        pr_obj = github.get_pull_request_by_number(pr)
+        if pr_obj is None:
+            app.display_error(f'Pull request {pr} not found')
+            return
+        file_list = github.get_changed_files_by_pr(pr_obj)
+    elif sha:
+        github = app.github
+        files_by_sha = github.get_changed_files_by_sha(sha)
+        if files_by_sha is None:
+            app.display_error(f'Commit {sha} not found')
+            return
+        file_list = files_by_sha
+    else:
+        app.display_error('No files provided')
+        return
+    if per_file:
+        owners_per_file = map_files_to_teams(file_list, codeowners)
+        app.display_info(owners_per_file)
+    else:
+        owners = get_owners(file_list, codeowners)
+        app.display_info(sorted(owners))
```

Comment:
```suggestion
    if per_file:
        owners_per_file = get_owners_by_file(file_list, codeowners)
        app.display_info(owners_per_file)
    else:
        owners = get_all_owners(file_list, codeowners)
        app.display_info(sorted(owners))
```
suggestion: in order to stay consistent with the `get_owners` function name, I think it's better to rename the `map_files_to_teams ` function to follow the same naming, since both functions are doing similar kind of work.